docs/docs/icechunk-python/storage.md

```diff
@@ -0,0 +1,229 @@
+        region='us-east-1',
+        access_key_id='my-access-key',
+        secret_access_key='my-secret-key',
+        # session token is optional
+        session_token='my-token',
+        endpoint_url=None,  # if using a custom endpoint
+        allow_http=False,  # allow http connections (default is False)
+    )
+    ```
+
+=== "Anonymous"
+
+    With this option, you connect to S3 anonymously (without credentials).
+    This is suitable for public data. [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        region='us-east-1,
+        anonymous=True,
+    )
+    ```
+
+=== "Refreshable Credentials"
+
+    With this option, you provide a callback function that will be called to obtain S3 credentials when needed. This is useful for workloads that depend on retrieving short-lived credentials from AWS or similar authority, allowing for credentials to be refreshed as needed without interrupting any workflows. [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    def get_credentials() -> S3StaticCredentials:
+        # In practice, you would use a function that actually fetches the credentials and returns them
+        # along with an optional expiration time which will trigger this callback to run again
+        return icechunk.S3StaticCredentials(
+            access_key_id="xyz",
+            secret_access_key="abc",å
+            expires_after=datetime.now(UTC) + timedelta(days=1)
+        )
+
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        region='us-east-1',
+        get_credentials=get_credentials,
+    )
+    ```
+
+#### AWS Limitations
+
+- Icechunk is currently incompatible with [S3 Express One Zone](https://aws.amazon.com/s3/storage-classes/express-one-zone/). See [this issue](https://github.com/earth-mover/icechunk/issues/667) for more discussion.
+
+#### Tigris
+
+[Tigris](https://www.tigrisdata.com/) is available as a storage backend for Icechunk. Functionally this storage backend is the same as S3 storage, but with a different endpoint. Icechunk provides a helper function specifically for [creating Tigris storage configurations](./reference.md#icechunk.tigris_storage). There are a few things to be aware of when using Tigris:
+
+- The `endpoint_url` parameter must be set to the Tigris endpoint. This is defaulted to `https://fly.storage.tigris.dev` but may be different depending on your Tigris configuration.
+- Tigris is a globally distributed object store by default. The caveat is that Tigris does not currently support the full consistency guarantees when the store is distributed across multiple regions. For now, to get the all the consistency guarantees Icechunk offers, you will need to setup your Tigris bucket as restricted to a single region. This can be done by setting the region in the Tigris bucket settings:
```

Comment:
```suggestion
- Tigris is a globally distributed object store by default. The caveat is that Tigris does not currently support the full consistency guarantees when the store is distributed across multiple regions. For now, to get all the consistency guarantees Icechunk offers, you will need to setup your Tigris bucket as restricted to a single region. This can be done by setting the region in the Tigris bucket settings:
```